Added in grid view in filebrowser

bertmeeuws opened the pull request:
@bertmeeuws
Refactoring code, and styling the contact add for visual consistency #… 

Fixed alignment

New changes since you last viewed
Adding hovers, visual enhancements and UX #123 #122 #124 #128

Added grid view for shared with me

added empty state in grid view

Grid in local filebrowser WOIP

Added a shared with me placeholder

Fixing state error


- Add grid view with empty state for shared with me & local
- Added warnings for example if a user uses a '/' in a folder name.
- Restyled the modals for visual consistency
- Added some hovers on icons

JonasWijne commented on index.html:
```diff
@@ -4,23 +4,19 @@
         <meta http-equiv="X-UA-Compatible" content="IE=edge" />
         <meta name="viewport" content="width=device-width,initial-scale=1.0" />
-        <link rel="icon" href="/logo_no_stroke.png"/>
+        <link rel="icon" href="/logo_no_stroke.png" />
```
weird

JonasWijne replied:
?

JonasWijne commented on public/config/version.ts:
```diff
@@ -1 +1 @@
-export default "{{version}}";
\ No newline at end of file
+export default '{{version}}';
```
```suggestion
export default "{{version}}";
```

JonasWijne replied:
not necessary

JonasWijne commented on .prettierrc:
```diff
@@ -1,10 +1,10 @@
-  "proseWrap": "preserve",
-  "printWidth": 120,
-  "vueIndentScriptAndStyle": true
+    "trailingComma": "es5",
```
whitespace => pls change

JonasWijne commented on README.md:
```diff
@@ -4,6 +4,6 @@ Experience as build on top of the digital twin (see twin_server repo).
-- chat interface
-- file manager interface
-- launcher video conferencing
+-   chat interface
```
whitespace

JonasWijne commented on index.html:
```diff
@@ -4,23 +4,19 @@
-            href="https://use.fontawesome.com/releases/v5.1.1/css/all.css"
-            rel="stylesheet"
-        />
+        <link href="https://use.fontawesome.com/releases/v5.1.1/css/all.css" rel="stylesheet" />
```
?

JonasWijne commented on index.html:
```diff
@@ -4,23 +4,19 @@
     <body class="h-full">
         <noscript>
             <strong
-                >We're sorry but <%= htmlWebpackPlugin.options.title %> doesn't
```
?

JonasWijne commented on index.html:
```diff
@@ -4,23 +4,19 @@
-                work properly without JavaScript enabled. Please enable it to
-                continue.</strong
+                >We're sorry but <%= htmlWebpackPlugin.options.title %> doesn't work properly without JavaScript
+                enabled. Please enable it to continue.</strong
```
?

JonasWijne commented on public/config/config.js:
```diff
@@ -1,5 +1,5 @@
-import config from '/config/config.json' assert { type: "json" };
+import config from '/config/config.json' assert { type: 'json' };
```
?

JonasWijne commented on public/config/config.js:
```diff
@@ -1,5 +1,5 @@
+import config from '/config/config.json' assert { type: 'json' };
 
-config.baseUrl =  `${window.location.origin}/`
+config.baseUrl = `${window.location.origin}/`;
```
?

JonasWijne commented on public/config/config.js:
```diff
@@ -1,5 +1,5 @@
 
-window.config = config
\ No newline at end of file
+window.config = config;
```
?

JonasWijne commented on public/config/config.json:
```diff
@@ -3,5 +3,4 @@
     "documentServerLocation": "https://documentserver.digitaltwin-test.jimbertesting.be/",
     "giphyApiKey": "uk3XRSO0vYrPDEQKDPZJ2wGz33qzIxST",
     "beta": true
-
```
unnecessary

JonasWijne commented on src/assets/index.css:
```diff
@@ -22,7 +22,7 @@
         white-space: nowrap;
     }
-    th{
+    th {
```
?

JonasWijne commented on src/components/AudioPlayer.vue:
```diff
@@ -1,39 +1,54 @@
 <template>
-    <div class='flex items-center p-2 pb-6'>
-        <audio ref='audiofile' :src='src' class='hidden' preload='auto'></audio>
+    <div class="flex items-center p-2 pb-6">
```
what is standard: make ticket to check all files follow up with project lead

JonasWijne commented on src/components/AudioPlayer.vue:
```diff
@@ -1,39 +1,54 @@
-            type='button'
-            @click='isPlaying = false'
+            v-if="isPlaying"
+            class="
```
what is standard: make ticker to check all files

JonasWijne commented on src/components/AudioPlayer.vue:
```diff
@@ -1,39 +1,54 @@
         >
-            <PauseIcon aria-hidden='true' class='h-5 w-5' />
-            <span class='sr-only'>Pause</span>
+            <PauseIcon aria-hidden="true" class="h-5 w-5" />
```
?

JonasWijne commented on src/components/AudioPlayer.vue:
```diff
@@ -1,39 +1,54 @@
-            class='ml-0.5 bg-white p-1.5 rounded-full text-gray-400 hover:text-accent-400 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-accent-500'
-            type='button'
-            @click='isPlaying = true'
+            class="
```
?

JonasWijne commented on src/components/AudioPlayer.vue:
```diff
@@ -1,39 +1,54 @@
+                focus:outline-none focus:ring-2 focus:ring-inset focus:ring-accent-500
+            "
+            type="button"
+            @click="isPlaying = true"
```
?

JonasWijne commented on src/components/Button.vue:
```diff
@@ -1,5 +1,7 @@
 <template>
-    <button class="flex items-center rounded-xl m-2 bg-accent-300 text-white btn disabled:bg-gray-300 bg-cursor-[unset]">
+    <button
```
why did you add this file please make sure you do git add with `git add -p`

JonasWijne commented on src/components/ChatCard.vue:
```diff
@@ -5,39 +5,49 @@
+        @keyup.enter="$emit('selectChat')"
     >
-        <div class='place-items-center relative'>
+        <div class="place-items-center relative">
```
?

JonasWijne reviewed (commented):
this merge conflict is bad please checkl first before comming back to me

JonasWijne commented:
commit messages not specific enough

bertmeeuws commented:
- Adding hovers on sidebars and modals
- Added cleaner looking exit button in modal
- Clicking outside of the modal will exit it
- ESCAPE key in modal will also exit it
- Using a "/" in a folder name will give real time feedback
- Added an error when a has no name
- Added a max. char length of 50 in a folder name
- When a file is now dragged into the fileuploader it doesn't reset, it takes the exisiting files in the array and adds the new one
- Put in error saying you need atleast one file when trying to upload if none are chosen
- Added a small fix for when /quantum is loaded but the directory isn't set to '/'
- Added grid mode for local filebrowser and shared with me folder
- Refactored the navigation in ContactAdd.vue to reading it out from an array and not hardcoded.
- You can now unselect folders/files by clicking the gray area

![image](https://user-images.githubusercontent.com/56439373/136036632-fc1aa48c-1018-4890-978c-78e5c9f7fd99.png)

![image](https://user-images.githubusercontent.com/56439373/136036722-e89c7b09-d655-425d-b5bb-36014c977c53.png)

![image](https://user-images.githubusercontent.com/56439373/136036753-18fd5676-d09d-439a-9c9b-f933cbf6512c.png)

![image](https://user-images.githubusercontent.com/56439373/136037315-cff65b9c-8b14-42e1-9ed1-6918fc9224aa.png)

JonasWijne reviewed (commented):
too much unnecessary changes

bertmeeuws commented on src/components/ChatCard.vue:
```diff
@@ -5,39 +5,49 @@
-        tabindex='0'
-        @click='$emit("selectChat")'
-        @keyup.enter='$emit("selectChat")'
+        class="
```
Added a hover & duration

bertmeeuws commented on src/components/ChatList.vue:
```diff
@@ -22,7 +22,18 @@
+                            bg-primary
+                            hover:bg-accent-800
+                            transition
+                            duration:300
```
Added a hover & duration

bertmeeuws commented on src/components/ContactAdd.vue:
```diff
@@ -2,23 +2,20 @@
             <a
-                :class="{ active: isActive('user') }"
-                class="nav-link grid-cols-6 text-center py-2"
+                v-for="item in navigation"
```
Navigation is now read out from an array

bertmeeuws commented on src/components/ContactAdd.vue:
```diff
@@ -170,12 +167,27 @@
+                        bg-primary
+                        hover:bg-accent-700
+                        transition
+                        duration-300
```
Added a hover & duration

bertmeeuws commented on src/components/Dialog.vue:
```diff
@@ -22,9 +24,25 @@
+                        class="
+                            w-6
+                            h-6
+                            bg-transparent
```
- Added a circle around the exit button in a modal
- Added a hover & duration

bertmeeuws commented on src/components/Dialog.vue:
```diff
@@ -37,7 +55,18 @@
+                            bg-primary
+                            hover:bg-accent-700
+                            transition
+                            duration-300
```
Added a hover & duration

bertmeeuws commented on src/components/Sidebar.vue:
```diff
@@ -15,7 +15,7 @@
                 :key="app.name"
                 :class="{
-                    'cursor-pointer': app.enabled,
+                    'cursor-pointer transition duration-300 hover:bg-primarylight rounded-xl': app.enabled,
```
Added a hover & duration

bertmeeuws commented on src/components/UserTable.vue:
```diff
@@ -67,7 +67,17 @@
+                                                bg-primary
+                                                hover:bg-accent-700
+                                                transition
+                                                duration-300
```
Added a hover & duration

bertmeeuws commented on src/components/UserTableGroup.vue:
```diff
@@ -64,7 +64,17 @@
+                                                    justify-self-end
+                                                    bg-primary
+                                                    hover:bg-accent-700
+                                                    transition
```
Added a hover & duration

bertmeeuws commented on src/components/fileBrowser/FileTable.vue:
```diff
@@ -1,17 +1,21 @@
+                    <ViewSelect />
+                </div>
+                <div class="overflow-hidden border-b border-gray-200 sm:rounded-lg">
+                    <FileDropArea @click.stop class="h-full" @send-file="uploadFiles">
```
Added a @click.stop to propagate the "deselect all files by clicking the gray background" when clicking around in the filebrowser itself

bertmeeuws commented on src/components/fileBrowser/FileTable.vue:
```diff
@@ -182,6 +185,130 @@
                                 </tr>
                             </tbody>
                         </table>
+                        <ul
```
This is the new grid view

bertmeeuws commented on src/components/fileBrowser/FileTable.vue:
```diff
@@ -308,6 +437,10 @@
                 isDraggingFiles.value = false;
             };
 
+            const goBack = () => {
```
When in grid view and a folder is empty you get the option to go back one page

bertmeeuws commented on src/components/fileBrowser/MainActionButtons.vue:
```diff
@@ -1,17 +1,26 @@
-        <button @click="showCreateFileDialog = true" class="text-white py-2 px-4 mr-2 rounded-md bg-primary">
-            <i class="fas fa-plus"></i> Upload Files
+        <button
+            @click="showCreateFileDialog = true"
```
Added a hover & duration

bertmeeuws commented on src/components/fileBrowser/MainActionButtons.vue:
```diff
@@ -1,17 +1,26 @@
     <div class="mx-2" :class="{ hidden: sharedDir }">
-        <button @click="showCreateFolderDialog = true" class="text-white py-2 px-4 mr-2 rounded-md bg-primary">
+        <button
+            @click="showCreateFolderDialog = true"
```
Added a hover & duration

bertmeeuws commented on src/components/fileBrowser/MainActionButtons.vue:
```diff
@@ -95,18 +107,34 @@
+                if (manualContactAdd.value.includes('/')) {
+                    createFolderErrors.value.push("'/' are not allowed in folder names.");
+                }
+                if (manualContactAdd.value.length >= 50) {
```
- Added max. char length in folders and it checks for '/'

bertmeeuws commented:
@JonasWijne Please sir, review my code

JonasWijne commented on src/store/fileBrowserStore.ts:
```diff
@@ -64,6 +69,7 @@ export const selectedAction = ref<Action>(Action.COPY);
 export const sharedDir = ref(false);
 export const sharedContent = ref<SharedFileInterface[]>([]);
 export const allSharedContent = ref<SharedFileInterface[]>([]);
+export const fileBrowserTypeView = ref<string>('LIST');
```
should be enum
```suggestion
export const fileBrowserTypeView = ref<View>('LIST');
```

JonasWijne commented on src/store/fileBrowserStore.ts:
```diff
@@ -38,6 +38,11 @@ export enum Action {
 }
 
+export enum View {
+    LIST,
```
```suggestion
    LIST = "LIST",
```
